api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -50,4 +51,37 @@ message RedisProxy {
 
   // Network settings for the connection pool to the upstream cluster.
   ConnPoolSettings settings = 3 [(validate.rules).message.required = true];
+
+  message RouteMatch {
```

Comment:
Can you potentially add arch level docs on this feature? That would help review the feature goals, etc. https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/redis

This feature will also need a release note.